lib/FieldLocation/FieldLocationFinalContainer.js

```diff
@@ -0,0 +1,89 @@
+import React, {
+  useState,
+  useCallback,
+  useEffect,
+} from 'react';
+import PropTypes from 'prop-types';
+import { uniqBy } from 'lodash';
+
+import { stripesConnect } from '@folio/stripes/core';
+import {
+  Loading,
+} from '@folio/stripes/components';
+
+import {
+  locationsManifest,
+} from '../manifests';
+import {
+  batchFetch,
+  getLocationOptions,
+} from '../utils';
+import FieldLocationFinal from './FieldLocationFinal';
+
+const FieldLocationFinalContainer = ({
+  labelId,
+  locationIds,
+  locationLookupLabel,
+  mutator,
+  name,
+  onChange,
+  required,
+}) => {
+  const [locationsForSelect, setLocationsForSelect] = useState();
+
+  useEffect(() => {
+    setLocationsForSelect();
+
+    batchFetch(mutator.fieldLocations, locationIds)
+      .then(locationsResponse => setLocationsForSelect(getLocationOptions(locationsResponse)))
+      .catch(() => setLocationsForSelect([]));
+  },
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  []);
+
+  const selectLocation = useCallback(
```

Comment:
if this is location fro plugin maybe it's better to give name more clear. for now there are `selectLocation` and `locationsForSelect` around, which is kinda confusing, at least for me